Singularity/Singularity/Screen/Button.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+using Singularity.Input;
+using Singularity.Property;
+
+
+namespace Singularity.Screen
+{
+    /// <summary>
+    /// Button class allows buttons to be created using either
+    /// text or Texture2D. The buttons can send out Clicked, Released,
+    /// and Hovering events that other classes can subscribe to.
+    /// </summary>
+    class Button : IUpdate, IDraw //TODO this needs to implement IWindowItem which will have update and draw
+    {
+        // bool to indicated if the button is made of text
+        private readonly bool mIsText;
+
+        private readonly float mScale;
+        private readonly Texture2D mButtonTexture;
+        private readonly string mButtonText;
+        private readonly Vector2 mPosition;
+        private readonly int mWidth;
+        private readonly int mHeight;
+        private readonly SpriteFont mFont;
+
+        // distinguish between mouse over hover or not
+        private Color mColor;
+
+        private Rectangle mBounds;
+        private bool mClicked;
+
+        // events that can be sent out by buttons
+        // an event handler must be created in the class that 
+        // has an instance of buttons is created in
+        // order to receive these events
+        public event EventHandler ButtonReleased;
```

Comment:
can you further specify how the events happen / are being used? I would not know how I could use such an event.

Reply:
id ask anyone with questions to contact me or look online since documentation on how to use this can take up quite some space

Reply:
fair enough. Add a link for refence please? (in comment)